Apps/Examples/Examples/All Examples/Annotations/PointAnnotationClusteringExample.swift

```diff
@@ -0,0 +1,105 @@
+        let mapInitOptions = MapInitOptions(cameraOptions: cameraOptions, styleURI: .light)
+        mapView = MapView(frame: view.bounds, mapInitOptions: mapInitOptions)
+        mapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+
+        view.addSubview(mapView)
+
+        // Add the source and style layers once the map has loaded.
+        mapView.mapboxMap.onNext(event: .mapLoaded) { _ in
+            self.addPointAnnotations()
+        }
+    }
+
+    func addPointAnnotations() {
+        // The image named `fire-station-11` is included in the app's Assets.xcassets bundle.
+        let image = UIImage(named: "fire-station-11")!
+        // Fire_Hydrants.geojson contains information about fire hydrants in the District of Columbia.
+        // It was downloaded on 6/10/21 from https://opendata.dc.gov/datasets/DCGIS::fire-hydrants/about
+        // Decode the GeoJSON into a feature collection on a background thread
+        _ = Bundle.main.url(forResource: "Fire_Hydrants", withExtension: "geojson")!
+        DispatchQueue.global(qos: .userInitiated).async {
+            guard let featureCollection = try? self.decodeGeoJSON(from: "Fire_Hydrants") else {
+                return
+            }
+
+            // Create an array of annotations for each fire hydrant
+            var annotations = [PointAnnotation]()
+            for feature in featureCollection.features {
+                guard let geometry = feature.geometry, case let Geometry.point(point) = geometry else {
+                    return
+                }
+                var pointAnnotation = PointAnnotation(coordinate: point.coordinates)
+                pointAnnotation.image = .init(image: image, name: "fire-station-11")
+                annotations.append(pointAnnotation)
+            }
+            DispatchQueue.main.async {
+                self.createClusters(annotations: annotations)
+            }
+        }
+    }
+
+    func createClusters(annotations: [PointAnnotation]) {
+        // Use a step expressions (https://docs.mapbox.com/mapbox-gl-js/style-spec/#expressions-step)
+        // with three steps to implement three sizes of circles:
+        //   * 15 when point count is less than 50
+        //   * 20 when point count is between 50 and 100
+        //   * 25 when point count is greater than or equal to 100
+        let circleRadiusExpression = Exp(.step) {
+            Exp(.get) {"point_count"}
+            15
+            50
+            20
+            100
+            25
+        }
+
+        // Use color levels to implement three colors of circles:
+        //   * green when point count is less than 50
+        //   * cyan when point count is between 50 and 100
+        //   * red when point count is greater than or equal to 100
+        let colorLevels = [
+            (pointCount: 100, clusterColor: StyleColor(.red)),
+            (pointCount: 50, clusterColor: StyleColor(.cyan)),
+            (pointCount: 0, clusterColor: StyleColor(.green))]
```

Comment:
Why define these as pairs, can we use an expression same as for circle radius above? So it would become:
```swift
let circleColorExpression = Exp(.step) {
    Exp(.get) { "point_count" }
    UIColor.green
    50
    UIColor.cyan
    100
    UIColor.red
}
```

With circle color option defined like: `var circleColor: Value<StyleColor>`

Then point annotation manager could have just one cluster circle layer(instead of each per level) with color passed from options: `circleLayer.circleColor = clusterOptions.circleColor`.

Reply:
I took this implementation from Android's `ClusterOptions` which has a `ColorLevels` pair: https://github.com/mapbox/mapbox-maps-android/blob/cbe8caedd8f2f9181b4f5fcac73cccca509f2fc1/sdk-base/src/main/java/com/mapbox/maps/plugin/annotation/ClusterOptions.kt#L80. @tobrun do you have any context on this design decision? 

@evil159 - I think your recommended approach works well and doesn't create unnecessary layers. My most recent commit takes this approach and is a bit cleaner. Should we go with this or stick more closely to the Android implementation?